### DISCUSSION TITLE
Ingress of Synb0-DISCO corrected data into qsiprep

### DISCUSSION THREAD

**Discussion by cookpa** [PennLINC/qsiprep#583] [Q&A]
I know there's plans to integrate Synb0 into qsiprep (#546), but I'm not sure what the timeline is on that, so I'm considering alternative approaches. 

Option 1: Run Synb0 and topup, and then run qsiprep with the anat and topup-corrected dwi data, and `--hmc-model eddy`. This involves an extra interpolation of the DWI data.

Option 2: Run Synb0 and get the undistorted b0. Copy this to fmap/ with a sidecar json that is identical to the dwi except:

  * PhaseEncodeDirection="j" (opposite to the dwi) 
  * TotalReadOutTime=0.0000001
  * EffectiveEchoSpacing = 0.0 (BIDS requires TotalReadOutTime > EffectiveEchoSpacing)

I'm trying out option 2 right now. Does this seem reasonable? 

Thanks for any suggestions! The use case is mostly b=1000 retrospective data (no field maps) and a bit of ABCD-style data (field maps incorrectly acquired and unusable)

**Comment by mattcieslak** [comment]
Hi Phil, option 2 will definitely work! It has a couple other nice advantages like eddy will be able to incorporate the fieldmap into its motion correction. Have you tried this out yet?

**Reply by cookpa** [reply]
Thanks Matt! I'm running a test case right now, I'll let you know how it turns out

**Reply by mattcieslak** [reply]
The only potential gotcha I can think of is that the visual report might look strange. I can't remember offhand how qsiprep picks which PE direction to show as the "before" image, but if it picks your synb=0 image the report won't have the same satisfying "move into place" effect. It will still be a useful visual check though, since the corrected image should be nicely registered to the synb=0 image

**Reply by cookpa** [reply]
You're right, it chose the synb0 image - results look pretty decent though

**Reply by mattcieslak** [reply]
this is great news!

**Reply by cookpa** [reply]
The report uses the first volume in imain returned by [get_best_b0_topup_inputs_from](https://github.com/PennLINC/qsiprep/blob/8ee6440798d2ea3010cb86c6b7e83b045813770a/qsiprep/interfaces/epi_fmap.py#L160).

One way to deal with this would be for that function to look for the largest TotalReadOutTime and return that one explicitly as the "before" image. PR?

**Reply by mattcieslak** [reply]
I was thinking about doing what we currently do for drbuddi, which is to make one visual report for plus and one for minus. but I like your idea as a way to get better reports right now

**Reply by cookpa** [reply]
Update: user error, the code is fine. It checks that the chosen "before" image matches distortion parameters with the dwi series. It is looking good now

**Reply by ShoujunYu** [reply]
> Update: user error, the code is fine. It checks that the chosen "before" image matches distortion parameters with the dwi series. It is looking good now

Hi, cookpa,

I currently have only the “before” image in the desc-sdc_b0.svg file, and it appears completely black. The distortion-corrected result is also not satisfactory.
My dataset contains only single-polar DWI, with just one b0 image, while all other volumes are high-b-value images. 

Could you advise on the proper way to use --use-syn-sdc? Thank you

**Reply by cookpa** [reply]
Hi, it's been quite a while since this issue so I would advise opening a new issue with full details of the data, what command you ran, and what the errors are. 

I can tell you that the `--use-syn-sdc` is not the same as SynB0. The name is confusing. The `--use-syn-sdc' uses SyN symmetric normalization, which predates SynB0 (synthesized b0), which is based on deep learning.

The way I have successfully used SynB0 is to run it offline, then ingress the data into qsiprep. As outlined above:

1. Run Synb0 and get the undistorted b0. 
2. Copy undistorted b0 to fmap/ with a sidecar json that is identical to the dwi except:

PhaseEncodeDirection="j" (opposite to the dwi, change this to be opposite to your data)
TotalReadOutTime=0.0000001
EffectiveEchoSpacing = 0.0 (BIDS requires TotalReadOutTime > EffectiveEchoSpacing)

Then I run qsiprep as I would for a "real" RPE b0 in fmap.

**Reply by ShoujunYu** [reply]
Thank you for providing the method. I was able to run it successfully, and the SDC results visually reasonable. Actually, I meet the similar problem as these two opening issues, [Issue #945](https://github.com/PennLINC/qsiprep/issues/945) and 
[Issue #930](https://github.com/PennLINC/qsiprep/issues/930), but I am considering to provide more details of my problems later.

I also noticed that there is a synb0 folder in the container image (qsiprep-1.0.2). Have you considered integrating synb0 into qsiprep, even though this method wasn’t reported in the original qsiprep paper? Thanks again!

**Reply by Alexis-Melot** [reply]
Hello, 
I am following the same methodology here : Synb0 then QSIPrep. 
But I struggle to know which output of synb0 to select and place in the fmap/ directory. Should it be b0_u.nii.gz or b0_all_topup.nii.gz ? 
I am running QSIPrep without fancy flags (except unringing method activated). Therefore, topup run along eddy. So, it seems to me it's not relevant to run topup with Synb0 (and it could save me some time) and I could go with passing b0_u.nii.gz file. Would it be a good way ?
